src/Native/SymSgdNative/SymSgdNative.cpp

```diff
@@ -0,0 +1,440 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// SymSGDNative.cpp : Defines the exported functions for the DLL application.
+
+#include <vector>
+#include <random>
+#if defined(USE_OMP)  
+#include <omp.h>
+#endif
+#include <unordered_map>
+#include "../Stdafx.h"
+#include "Macros.h"
+#include "SparseBLAS.h"
+#include "SymSgdNative.h"
+
+// This method learns for a single instance
+inline void LearnInstance(int instSize, int* instIndices, float* instValues,
+	float label, float alpha, float l2Const, float piw, float& weightScaling, float* weightVector, float& bias) {
```

Comment:
(nit) It would be nice to follow the coding style already established in the repo and put the opening `{` on a new line:

```C++
inline void LearnInstance(int instSize, int* instIndices, float* instValues,
 float label, float alpha, float l2Const, float piw, float& weightScaling, float* weightVector, float& bias)
{
    //code
}
```